django/test/testcases.py

```diff
@@ -595,6 +595,12 @@ def assertFormError(self, response, form, field, errors, msg_prefix=""):
         for i, context in enumerate(contexts):
             if form not in context:
                 continue
+            if not context[form].is_bound:
+                form_repr = repr(context[form])
+                self.fail(
+                    f"{msg_prefix}The form {form_repr} is not bound, it will never "
+                    f"have any errors."
+                )
```

Comment:
Might be less clunky putting the form in a local variable:
```suggestion
            if form_obj := context.get(form):
                continue
            if not form_obj.is_bound:
                self.fail(
                    f"{msg_prefix}The form {form_obj!r} is not bound, it will never "
                    f"have any errors."
                )
```

`context[form]` is repeated an awful lot further down too. You'll probably want a better name than `form_obj` though... 🤔

Reply:
I didn't want to change this because it's just one of subsequent changes from #15179. Finally, `context[form]` will not be used in new methods.

Reply:
Ah, I hadn't looked at #15179 in detail. Even better! 😁